R/mock-object.R

```diff
@@ -0,0 +1,81 @@
+
+#' Create a mock object.
+#'
+#' Mock object's primary use is to record calls that are made on the
+#' mocked function. Optionally values can be passed via \code{...} to
+#' make the mock object return them upon subsequent calls. If no value
+#' is passed in \code{...} then \code{NULL} is returned.
+#'
+#' @param ... Values returned upon subsequent calls.
+#' @param cycle Whether to cycle over the return values. If \code{FALSE},
+#'        will fail if called too many times.
+#' @export
+#' @examples
+#' \dontrun{
+#'   m <- mock(1)
+#'   with_mock(summary = m, {
+#'     expect_equal(summary(iris), 1)
+#'     expect_call(m, 1, summary(iris))
+#'   })
+#' }
+#'
+mock <- function (..., cycle = FALSE) {
+  returns <- list(...)
```

Comment:
You could also say `returns <- eval(substitute(alist(...)))` and then use `eval(returns[[...]])` when evaluating the mocks. This will make sure the expressions are evaluated only when the mock is called, not at the start of the function.